server/sonar-server/src/test/java/org/sonar/server/computation/task/projectanalysis/component/ComponentRootBuilderTest.java

```diff
@@ -466,18 +470,37 @@ public void createFileAttributes_returns_null_when_type_is_not_FILE() {
 
   @Test
   public void createFileAttributes_sets_language_to_null_when_unset_in_Scanner_Component() {
-    assertThat(createFileAttributes(newBuilder().setType(FILE).build()).getLanguageKey()).isNull();
+    assertThat(createFileAttributes(newBuilder().setType(FILE).setLines(1).build()).getLanguageKey()).isNull();
   }
 
   @Test
   public void createFileAttributes_sets_language_to_null_when_empty_in_Scanner_Component() {
-    assertThat(createFileAttributes(newBuilder().setType(FILE).setLanguage("").build()).getLanguageKey()).isNull();
+    assertThat(createFileAttributes(newBuilder().setType(FILE).setLanguage("").setLines(1).build()).getLanguageKey()).isNull();
   }
 
   @Test
   public void createFileAttributes_sets_unitTest_from_Scanner_Component() {
-    assertThat(createFileAttributes(newBuilder().setType(FILE).build()).isUnitTest()).isFalse();
-    assertThat(createFileAttributes(newBuilder().setType(FILE).setIsTest(true).build()).isUnitTest()).isTrue();
+    assertThat(createFileAttributes(newBuilder().setType(FILE).setLines(1).build()).isUnitTest()).isFalse();
+    assertThat(createFileAttributes(newBuilder().setType(FILE).setIsTest(true).setLines(1).build()).isUnitTest()).isTrue();
+  }
+
+  @Test
+  public void createFileAttributes_sets_lines_in_Scanner_Component() {
+    assertThat(createFileAttributes(newBuilder().setType(FILE).setLines(10).build()).getLines()).isEqualTo(10);
+  }
+
+  @Test
+  public void fail_with_IAE_when_createFileAttributes_sets_lines_to_0() throws Exception {
```

Comment:
Yes it is ! I've already checked with scanner team, and even empty file have a number of lines to 1